core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -511,36 +511,45 @@ private void processData(Map<String, Set<String>> inputStudySampleMap,
                 // "0" indicates Cases with both mutations and copy number alterations data
                 // "1" indicates Cases with mutations data
                 // "2" indicates Cases with copy number alterations data
-                List<String> predefinedCaseSetIds = Arrays.asList("0", "1", "2");
-                
-                if(predefinedCaseSetIds.contains(sampleSetId)) {
-                        dataTypePriority = Integer.parseInt(sampleSetId);
+                SampleListCategory _sampleListCategory = SampleListCategory.ALL_CASES_IN_STUDY;
+                String updatedSampleSetId = "all";
+                switch (sampleSetId) {
+	                case "0":
```

Comment:
Let's take the opportunity to support both `0`,`1`,`2` and `w_mut`, `w_cna`, `w_mut_cna`. And in the frontend pr, we can switch to `w_mut`, `w_cna`, `w_mut_cna`.